NEW : Add task categories 1/3

defrance opened the pull request:
first part of work to add categories on task

second part left is to implement use on task.php, tasks.php and viewcat.php

defrance committed:
Create llx_categorie_project_task table

defrance committed:
Add primary key and indexes to llx_categorie_project_task

defrance committed:
Add project_task type to category class

defrance committed:
Implement setCategories method in task class

Added setCategories method to manage task categories.

frederic34 commented:
@defrance can you make a separate PR with only new sql tables (and migration)

defrance commented:
they are no migration , just one new table and index

I have already split my PR into two because adding categories to the intervention took 2 years (refusal of a global PR then decomposition into 3 parts of the PR and finally pieces which were integrated into two different versions of dolibarr)

frederic34 commented:
If there is a new table, you need to put it in migration 22 -> 23 sql file (and travis will be happy)

defrance committed:
Add llx_categorie_project_task table with constraints

Create llx_categorie_project_task table and add constraints

defrance committed:
Update copyright and remove setCategories method

Updated copyright year for Charlene Benke and removed the setCategories method.

defrance committed:
Remove project_task constant and related mappings

defrance commented:
ok i have suppress php files (except indenting), and add sql to migrate

defrance committed:
Modify primary key for llx_categorie_project_task

Updated primary key constraint for llx_categorie_project_task table.

defrance commented on htdocs/install/mysql/migration/22.0.0-23.0.0.sql:
```diff
@@ -43,6 +43,20 @@ ALTER TABLE llx_opensurvey_user_studs ADD COLUMN tms timestamp DEFAULT CURRENT_T
+  import_key    	varchar(14)
+) ENGINE=innodb;
+
+ALTER TABLE llx_categorie_project_task ADD PRIMARY KEY pk_categorie_project (fk_categorie, fk_task);
```
ok just made it

defrance reviewed (commented):
ok

eldy commented on htdocs/install/mysql/tables/llx_categorie_project_task.key.sql:
```diff
@@ -0,0 +1,25 @@
+-- ============================================================================
```
Can you rename the file into
htdocs/install/mysql/tables/llx_categorie_project_task-project.key.sql
So the sql file should be executed only module project is enabled.

defrance committed:
Rename llx_categorie_project_task.key.sql to llx_categorie_project_task-project.key.sql

defrance committed:
Rename llx_categorie_project_task.sql to llx_categorie_project_task-project.sql

defrance reviewed (commented):
ok for the renaming

defrance committed:
Merge branch 'develop' into patch-820897

defrance committed:
Merge branch 'develop' into patch-820897
